modules/michaoBidAdapter.js

```diff
@@ -0,0 +1,352 @@
+import { ortbConverter } from '../libraries/ortbConverter/converter.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { config } from '../src/config.js';
+import { BANNER, NATIVE, VIDEO } from '../src/mediaTypes.js';
+import { Renderer } from '../src/Renderer.js';
+import {
+  deepSetValue,
+  isArray,
+  isBoolean,
+  isNumber,
+  isStr,
+  logError,
+  replaceAuctionPrice,
+  triggerPixel,
+} from '../src/utils.js';
+
+const ENV = {
+  BIDDER_CODE: 'michao',
+  SUPPORTED_MEDIA_TYPES: [BANNER, VIDEO, NATIVE],
+  ENDPOINT: 'https://rtb.michao-ssp.com/openrtb/prebid',
+  NET_REVENUE: true,
+  DEFAULT_CURRENCY: 'USD',
+  OUTSTREAM_RENDERER_URL:
+    'https://cdn.jsdelivr.net/npm/in-renderer-js@1.0.6/dist/in-video-renderer.umd.min.js',
+};
+
+export const spec = {
+  code: ENV.BIDDER_CODE,
+  supportedMediaTypes: ENV.SUPPORTED_MEDIA_TYPES,
+
+  isBidRequestValid: function (bid) {
+    const params = bid.params;
+
+    if (!isNumber(params?.site)) {
+      domainLogger.invalidSiteError(params?.site);
+      return false;
+    }
+
+    if (!isStr(params?.placement)) {
+      domainLogger.invalidPlacementError(params?.placement);
+      return false;
+    }
+
+    if (params?.partner) {
+      if (!isNumber(params?.partner)) {
+        domainLogger.invalidPartnerError(params?.partner);
+        return false;
+      }
+    }
+
+    if (params?.badv) {
+      if (!isArray(params?.badv)) {
+        domainLogger.invalidBadvError(params?.badv);
+        return false;
+      }
+    }
+
+    if (params?.bcat) {
+      if (!isArray(params?.bcat)) {
+        domainLogger.invalidBcatError(params?.bcat);
+        return false;
+      }
+    }
+
+    if (bid.params?.reward) {
+      if (!isBoolean(params?.reward)) {
+        domainLogger.invalidRewardError(params?.reward);
+        return false;
+      }
+    }
+
+    const video = bid.mediaTypes?.video;
```

Comment:
All these your checks are not appropriate in your adapter, the type check should have already happened somewhere else. If not, let's add it in that place so we don't have every adapter validating it's input duplicatively